### PR TITLE
Add a job to build 3.7 sources from git and release RPM

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Updated with version info

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/version_map.groovy [hunk 1]
```diff
@@ -0,0 +1,5 @@
+def versions = [
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/version_map.groovy]: We have a similar mapping in the katello job. Can we unify that in 1 file so we can't forget one?

**Comment by ehelms** [comment]
Updated -- please check the logic introduced for that update

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.7/rpmRelease.groovy [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Build Source') {
+            parallel {
+                stage('build-katello-installer-tarball') {
+                    steps {
+                        dir('katello-installer') {
+                            configureRVM('2.2')
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/katello/3.7/rpmRelease.groovy]: Should we just make this 2.4? Also in a variable so it's only in 1 place?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy [hunk 3]
```diff
@@ -16,7 +10,7 @@ pipeline {
                 ghprb_git_checkout()
 
                 dir('foreman') {
-                    git url: "https://github.com/theforeman/foreman", branch: katello_versions[ghprbTargetBranch]
+                    git url: "https://github.com/theforeman/foreman", branch: foremanVersion(ghprbTargetBranch)
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy]: This doesn't append the `-stable` part. The job definition probably also needs to include `version_map.groovy`

**Comment by ehelms** [comment]
Updated

**Comment by ekohl** [comment]
Given we already pushed an RC1, is this still wise to do?

**Comment by ehelms** [comment]
Wise? Absolutely. This should be valid for every release. The technological solution is the tough part.

**Comment by ekohl** [comment]
Given 3.7 is now EOL, should we close this or update it to the current version?

**Pull request closed without merging**